firmware/hw_layer/adc_subscription.cpp

```diff
@@ -27,7 +27,7 @@ void AdcSubscription::SubscribeSensor(FunctionalSensor &sensor,
 	}
 
 	// bounds check
-	if (s_nextEntry >= std::size(s_entries)) {
+	if (s_nextEntry >= 8) { 		/* std::size(s_entries) is invalid */
```

Comment:
What's wrong with `std::size`? It returns the number of elements in the array. It needs c++17, though.

Reply:
![image](https://user-images.githubusercontent.com/568254/68351529-99a60c00-00b8-11ea-92b5-04c36850df3c.png)